### ISSUE TITLE
Cannot import the third-party libraries

### ISSUE THREAD

**Issue by MarcSteven** [CocoaPods/CocoaPods#4769]
Yesterday I imported the library into my project via cocoapods.But it told me there's a error.the error displayed .I changed the setting of target
1,Click the target 
2,enter into the setting of Build Phases .and then -Enbed Framework...Selected the item "Run script only when installing".The order is over.The error disappeared completely....
I had confirmed with the writer of Alamofire.......He told me perhaps that's the issue of cocoa pods.

**Comment by Mqhong** [comment]
update your cocoapods version  And let your class  public

**Comment by MarcSteven** [comment]
Thank you so much,the progress had been done before creating the new issue.....Perhaps it's a bug of cocoa pods...I don't know why .

**Comment by neonichu** [comment]
We need more information to debug such an issue, like an example project. Please have a look at <https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md> for a lists of things that belong in an actionable issue.

**Comment by MarcSteven** [comment]
I meant I had installed the cocoa pods  completely.And then I had imported framework of Alamofire.But unfortunately sometimes I cannot import it successfully.So it caused more errors eg enum cannot conform protocol of URLRequestConvertible ......According to your reply I had checked it carefully...But still cannot solve it .I have a bit helpless and upset...Sorry to disturb your job....Thank you so much....

**Comment by neonichu** [comment]
Sorry, but this still doesn't tell us anything about how we might reproduce your issue. What steps did you perform from the point where your project worked as expected to the point where it did not anymore?

A common error is described in [this comment](https://github.com/CocoaPods/CocoaPods/issues/4756#issuecomment-170959531) — once you manually edit the linker flags, CocoaPods cannot change them for you anymore.

**Comment by MarcSteven** [comment]
Ok,the steps as below:
first of all I created a project....
second:open my shell and then build pod file  ,and then input the related framework .(The path is accurate ,so never care about it)
Third step:pod install and then update...
Fourth step:Open workspace.project....
The result is that 
In some files I can import framework of Alamofire .I think maybe that's key ......
Sorry to disturb you from time to time......If have errors ,pls point out timely.I'm not skilled in Cocoapods.Thank for your invention .Save time ...Mid night,go to bed ...See you tomorrow....

**Comment by MarcSteven** [comment]
Hi ,the issue is still existing ,so I uninstalled cocoa pods  manually .But I cannot accomplish to install cocoa pods  successfully.The error as below:
ERROR:  Could not find a valid gem 'cocoapods' (>= 0), here is why:
          Unable to download data from https://ruby.taobao.org - no such name (https://ruby.taobao.org/latest_specs.4.8.gz)
According to the information I guess maybe the link is wrong.Thank you so much....Sorry to disturb you again.I'm not skilled in this field of cocoa pods .But to my surprised it's convenient  to import the third-party libraries atomically  ......Thank you so much.....

**Comment by MarcSteven** [comment]
That's the information about the error.I have no method to solve it ....Hope get more people' help to solve it as soon as possible...Thank you so much...
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ gem install cocoapods
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.0.0 directory.
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ sudo gem install cocoapods
ERROR:  While executing gem ... (Errno::EPERM)
    Operation not permitted - /usr/bin/pod
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ gem install cocoapods
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.0.0 directory.
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ pod search
/Library/Ruby/Site/2.0.0/rubygems/dependency.rb:315:in `to_specs': Could not find 'cocoapods' (>= 0) among 33 total gem(s) (Gem::LoadError)
Checked in 'GEM_PATH=/Users/zhaoxinqiang/.gem/ruby/2.0.0:/Library/Ruby/Gems/2.0.0:/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/gems/2.0.0', execute `gem env` for more information
	from /Library/Ruby/Site/2.0.0/rubygems/dependency.rb:324:in `to_spec'
	from /Library/Ruby/Site/2.0.0/rubygems/core_ext/kernel_gem.rb:64:in `gem'
	from /usr/local/bin/pod:22:in `<main>'
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ gem install cocoapods --user-install
WARNING:  You don't have /Users/zhaoxinqiang/.gem/ruby/2.0.0/bin in your PATH,
	  gem executables will not run.
Successfully installed cocoapods-0.39.0
Parsing documentation for cocoapods-0.39.0
Installing ri documentation for cocoapods-0.39.0
1 gem installed
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ cocoapods uninstall
-bash: cocoapods: command not found
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ gem uninstall cocoapods
Remove executables:
	pod, sandbox-pod

in addition to the gem? [Yn]  y
Removing pod
Removing sandbox-pod
Successfully uninstalled cocoapods-0.39.0
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ ls -l /Library/Ruby/Gems/2.0.0/specifications/default
total 0
-rw-r--r--  1 root  wheel   767  8 23 08:06 bigdecimal-1.2.0.gemspec
-rw-r--r--  1 root  wheel   602  8 23 08:06 io-console-0.4.2.gemspec
-rw-r--r--  1 root  wheel   537  8 23 08:06 json-1.7.7.gemspec
-rw-r--r--  1 root  wheel   341  8 23 08:06 minitest-4.3.2.gemspec
-rw-r--r--  1 root  wheel  4837  8 23 08:06 psych-2.0.0.gemspec
-rw-r--r--  1 root  wheel  1481  8 23 08:06 rake-0.9.6.gemspec
-rw-r--r--  1 root  wheel  2758  8 23 08:06 rdoc-4.0.0.gemspec
-rw-r--r--  1 root  wheel   763  8  1 07:08 test-unit-2.0.0.0.gemspec
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ sudo mkdir /Library/Ruby/Gems/2.0.0/specifications/default
Password:
mkdir: /Library/Ruby/Gems/2.0.0/specifications/default: File exists
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ sudo curl -o /Library/Ruby/Gems/2.0.0/specifications/default/json-1.7.7.gemspec \
>                https://raw.githubusercontent.com/flori/json/v1.7.7/json.gemspec
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  4842  100  4842    0     0   1663      0  0:00:02  0:00:02 --:--:--  1662
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ gem install cocoapods
Fetching: cocoapods-0.39.0.gem (100%)
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.0.0 directory.
Marc-Stevens-MacBook-Pro:~ zhaoxinqiang$ sudo gem install cocoapods
ERROR:  While executing gem ... (Errno::EPERM)
    Operation not permitted - /usr/bin/pod

**Comment by orta** [comment]
You might find it easier to use the [CocoaPods Mac App](https://cocoapods.org/app) - it's really hard to follow this thread I'm afraid @MarcSteven

**Comment by MarcSteven** [comment]
Yeah I never use it .Is it easy to manager the third -party libraries ?It doesn't use command line to execute the order ,is it?I cannot bear the command line order to execute something.Too hard to use it. I meet more strange things ,very weird  ....

**Comment by MarcSteven** [comment]
@orta  But it cannot solve it as that's not root cause of issue...Looking forward to solving it as soon as possible,One week had gone,but no any useful information to help me to solve it .......I don't know what to do.Fuzzy rhythm  and very upset ...

**Comment by orta** [comment]
I don't really understand your problem though, that's why you were asked to read the contribution page.

Can you provide a re-producable project with your problem for us rather than explain it?

**Comment by MarcSteven** [comment]
Perhaps the issue is from Xcode...It just occurred  from time to time.At present I wanna confirm the root cause ,I uninstalled cocoa pods  and then wanna install it again.
But it cannot install successfully.I don't know what to do.The error as above .
I'm not skilled in the filed of cocoa pods and Unix.....So I need help....@orta

**Comment by orta** [comment]
The Mac app is the answer to that, you do not need to use the command line if you use the [Mac app](https://cocoapods.org/app). Otherwise you should look into the sudo-less guide for CocoaPods: https://guides.cocoapods.org/using/getting-started.html#sudo-less-installation

**Comment by MarcSteven** [comment]
But it's no any help .How to create pod file via mac App...I tried more methods but till now still invalid ........Perhaps I was too stupid.....

**Comment by orta** [comment]
You're definitely not too stupid. From the sounds of it, you've not read the guides or seen any of the tutorials how CP works. It's a file, you can make it in a text editor. 

We use these issues to keep track of bug reports and feature requests, and this issue veered out of that territory I'm afraid.
